addons/mail/models/res_users.py

```diff
@@ -83,15 +83,20 @@ def _create_welcome_message(self):
 
     @api.model
     def activity_user_count(self):
-        query = """SELECT m.id, count(*), act.res_model as model,
+        # Reminders don't having any model so here we assign 'mail.activity' as reminder's model.
+        query = """SELECT m.id, count(*),
+                        CASE
+                            WHEN act.res_model IS NULL Then 'mail.activity'
```

Comment:
Why not supporting res_model=False in JS and renaming it to Reminder instead of hardcoding it on mail.activity ? That way we keep the same model backend and frontend: void res_model = reminder. Meaning of this field stays the same.